File: src/main/java/com/web/dim_on2/domain/HitResult.java
package com.web.dim_on2.domain;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import jakarta.persistence.*;
import java.time.ZonedDateTime;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Embeddable
public class HitResult {

    @Id
    @Setter
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Embedded
    private Coordinate coords;

    @Embedded
    private ShotResult shot;

    private ZonedDateTime currentTime;


    public static HitResult fromHit(Coordinate coordinate, ShotResult shotResult) {
        return new HitResult(null, coordinate, shotResult, ZonedDateTime.now());
    }

    @Override
    public String toString() {
        return "HitResult{" +
                "id=" + id +
                ", Coordinates=" + coords +
                ", ShotResult=" + shot +
                '}';
    }
}

File: src/main/resources/react_app/src/util/HitResult.js
export class HitResult {
    constructor(x, y, r, isHit, currentTime, executionTime) {
        this._x = x;
        this._y = y;
        this._r = r;
        this._isHit = isHit;
        this._currentTime = currentTime;
        this._executionTime = executionTime;
    }

    get executionTime() {
        return this._executionTime;
    }
    get currentTime() {
        return this._currentTime;
    }
    get isHit() {
        return this._isHit;
    }
    get r() {
        return this._r;
    }
    get y() {
        return this._y;
    }
    get x() {
        return this._x;
    }
}